modules/wallet/transactions.go

```diff
@@ -89,19 +104,27 @@ func (w *Wallet) Transactions(startHeight, endHeight types.BlockHeight) (pts []m
 	if startHeight > w.consensusSetHeight || startHeight > endHeight {
 		return nil, errOutOfBounds
 	}
-	if len(w.processedTransactions) == 0 {
-		return nil, nil
-	}
 
-	for _, pt := range w.processedTransactions {
-		if pt.ConfirmationHeight > endHeight {
-			break
-		}
-		if pt.ConfirmationHeight >= startHeight {
-			pts = append(pts, pt)
+	err = w.db.View(func(tx *bolt.Tx) error {
+		c := tx.Bucket(bucketProcessedTransactions).Cursor()
+		for key, val := c.First(); key != nil; key, val = c.Next() {
+			var pt modules.ProcessedTransaction
+			if err := encoding.Unmarshal(val, &pt); err != nil {
+				return err
+			}
+			if pt.ConfirmationHeight < startHeight {
+				continue
+			} else if pt.ConfirmationHeight > endHeight {
+				// transactions are stored in chronological order, so we can
+				// break as soon as we are above endHeight
```

Comment:
ideally we would seek to `startHeight` first, but this is not possible because the key is a sequence integer, not a blockheight. Using a blockheight would mean storing a `[]Transaction` as the value rather than a single transaction. I'm not too worried about this though because this code will likely be obsoleted by caching anyway.